frontends/web/src/routes/account/receive/receive.tsx

```diff
@@ -15,376 +15,309 @@
+            if (addressIndex === -1) {
+                addressIndex = 0;
             }
+            setCurrentAddressIndex(addressIndex);
+            setCurrentAddresses(receiveAddresses[addressIndex].addresses);
         }
-    }
+    }, [addressType, availableScriptTypes, getIndexOfMatchingScriptType, receiveAddresses]);
 
-    private verifyAddress = (addressesIndex: number) => {
-        const { receiveAddresses, secureOutput, code } = this.props;
-        if (code === undefined) {
-            return;
-        }
-        const { activeIndex } = this.state;
-        if (!secureOutput.hasSecureOutput) {
-            this.unregisterEvents();
-            alertUser(this.props.t('receive.warning.secureOutput'), {
-                callback: this.registerEvents
-            });
-            return;
+    const verifyAddress = (addressesIndex: number) => {
+        if (receiveAddresses && secureOutput) {
+            if (code === undefined) {
+                return;
+            }
+            if (!secureOutput.hasSecureOutput) {
+                alertUser(t('receive.warning.secureOutput'));
+                return;
+            }
+            setVerifying(true);
+            accountApi.verifyAddress(code, receiveAddresses[addressesIndex].addresses[activeIndex].addressID)
+                .then(() => setVerifying(false));
         }
-        this.setState({ verifying: true });
-        accountApi.verifyAddress(
-            code,
-            receiveAddresses[addressesIndex].addresses[activeIndex].addressID).then(() => {
-                this.setState({ verifying: false });
-            });
-    }
+    };
 
-    private previous = (e: React.SyntheticEvent) => {
+    const previous = (e: React.SyntheticEvent) => {
         e.preventDefault();
-        const activeIndex = this.state.activeIndex;
-        if (!this.state.verifying && activeIndex > 0) {
-            this.setState({
-                activeIndex: activeIndex - 1,
-            });
+        if (verifying && activeIndex > 0) {
+            setActiveIndex(activeIndex - 1);
         }
-    }
+    };
 
-    private next = (e: React.SyntheticEvent, numAddresses: number) => {
+    const next = (e: React.SyntheticEvent, numAddresses: number) => {
         e.preventDefault();
-        const { verifying, activeIndex } = this.state;
         if (!verifying && activeIndex < numAddresses - 1) {
-            this.setState({
-                activeIndex: activeIndex + 1,
-            });
+            setActiveIndex(activeIndex + 1);
         }
-    }
+    };
 
-    private getAccount = () => {
-        return this.props.accounts.find(({ code }) => code === this.props.code);
-    }
+    // enable copying only after verification has been invoked if verification is possible and not optional.
+    const forceVerification = secureOutput && secureOutput.hasSecureOutput && !secureOutput.optional;
```

Comment:
now while `hasSecureOutput()` is not done yet, `forceVerification` is false. Before, the whole screen was not rendered.

In absence of a spinner, maybe defaulting to `true` is better, as that optimizes for the BitBox02?

Reply:
https://github.com/digitalbitbox/bitbox-wallet-app/pull/1668/commits/763d60138728f059e1f198851dbfcafe9e151a40

will squash that one too.

Reply:
But I added 1 more commit that moved the verifylabel and button into its own component, here https://github.com/digitalbitbox/bitbox-wallet-app/pull/1668/commits/5ee5c3655243f228cee0b323fed19f7a08c2f2bf

should I keep the commit or also squash?

PTAL